Gemfile

```diff
@@ -182,13 +183,12 @@ gem 'dynamic_form'
 gem 'record_tag_helper', '~> 1.0'
 
 group :test do
-  # To remove once migrated all functional tests
-  gem 'codecov', :require => false
   gem 'rack-no_animations', '~> 1.0.3'
   gem 'rails-controller-testing', '~> 1.0.4'
-  gem 'simplecov', '~> 0.21.2', require: false
+  gem 'simplecov', '~> 0.22.0', require: false
+  gem 'simplecov-cobertura', '~> 2.1'
 
-  gem 'capybara', '~>3.35.3', source: 'https://rubygems.org'
+  gem 'capybara', '~> 3.40.0'
```

Comment:
Fixing:
```
/home/dmayorov/.asdf/installs/ruby/3.1.4/lib/ruby/gems/3.1.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:17:in `require': cannot load such file -- matrix (LoadError)
```

Reply:
I think we shouldn't add to Gemfile intermittened dependencies. They may come and go.

Reply:
What do you mean by "intermittened dependencies" ? Do you think we need to downgrade some of the upgraded deps?

Reply:
Means dependencies of dependencies and not what we require on it's own. Do we use `matrix` gem directly? If not, then let the actual gems request their supported versions.

Reply:
> Means dependencies of dependencies and not what we require on it's own. Do we use `matrix` gem directly? If not, then let the actual gems request their supported versions.

We don't use it directly.
The long story is that `matrix` was a default gem in 2.7, but in Ruby 3.1 it's not anymore. So, this capybara upgrade fixes it, because later versions require `matrix` explicitly: https://github.com/teamcapybara/capybara/commit/a96cb7ccdcc193e4732e3464a76f21c0ae8a65e6

We could do the same for prawn. Our [current version](https://github.com/prawnpdf/prawn/blob/2.4.0/prawn.gemspec#L38) doesn't have it as a dependency because it was expected to be a default gem, however the latest 2.5.0 [does](https://github.com/prawnpdf/prawn/blob/2.5.0/prawn.gemspec#L50).

If you think it's better to upgrade `prawn` than add `matrix` explicitly, then we can try.

Reply:
Got it now. Maybe just add a comment in Gemfile that it is there only to workaround the prawn dependency. I don't think it is good to start upgrading everything at once. Ruby 3.1 is a big enough upgrade.